components/raftstore/src/store/peer_storage.rs

```diff
@@ -163,34 +164,30 @@ impl EntryCache {
     }
 
     fn append(&mut self, tag: &str, entries: &[Entry]) {
-        if entries.is_empty() {
-            return;
+        if !entries.is_empty() {
+            let mut mem_size_change = 0;
+            let old_capacity = self.cache.capacity();
+            self.append_impl(tag, entries);
+            let new_capacity = self.cache.capacity();
+            mem_size_change += Self::get_cache_vec_mem_size_change(new_capacity, old_capacity);
+            mem_size_change += self.shrink_if_necessary();
+            Self::flush_mem_size_change(mem_size_change);
```

Comment:
flush two times in one append?

Reply:
It's OK because `flush_mem_size_change` only changes one atomic number.

Reply:
So what is the purpose of flushing in append_impl?

Reply:
However it's better to change it to `fn append_impl(...) -> i64`.